lib/sequelize.js

```diff
@@ -455,121 +455,127 @@ class Sequelize {
    *
    * @see {@link Model#build} for more information about instance option.
    */
-  query(sql, options) {
-    if (arguments.length > 2) {
-      // TODO: Remove this note in the next major version (4.0)
-      throw new Error('Sequelize.query was refactored to only use the parameters `sql` and `options`. Please read the changelog about BC.');
-    }
 
-    options = _.assign({}, this.options.query, options);
+  query(sql, options) {
+    let queryArguments  = arguments;
+    let bindParameters;
+    return Promise.try(() => {
 
-    if (options.instance && !options.model) {
-      options.model = options.instance.constructor;
-    }
+      if (queryArguments.length > 2) {
+        // TODO: Remove this note in the next major version (4.0)
+        throw new Error('Sequelize.query was refactored to only use the parameters `sql` and `options`. Please read the changelog about BC.');
```

Comment:
As this targets master, this whole note can be removed and therefor the `arguments` check